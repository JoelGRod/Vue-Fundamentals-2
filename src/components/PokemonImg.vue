<template>
  <div class="pokemon-container">
    <img
      :src="getImgSrc"
      class="hidden-pokemon"
      alt="Pokemon Hide Image"
    />
    <img
      :src="getImgSrc"
      v-if="showPokemon"
      class="fade-in"
      alt="Pokemon Image"
    />
  </div>
</template>

<script>
export default {
  name: "PokemonImg",
  props: {
    imgId: {
      type: Number,
      required: true,
      default: 1
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  computed: {
    getImgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.imgId}.svg`
    }
  }
};
</script>

<style scoped>
.pokemon-container {
  height: 200px;
}
img {
  height: 200px;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>