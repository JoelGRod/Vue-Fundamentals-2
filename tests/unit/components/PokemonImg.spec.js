import { shallowMount } from '@vue/test-utils';
import PokemonImg from '@/components/PokemonImg';

describe('PokemonImg.vue', () => {
    test('should match with snapshot', () => {
        // Arrange
        const wrapper = shallowMount( PokemonImg, {
            props: {
                imgId: 1,
                showPokemon: true
            }
        });
        // Act
        // Assert
        expect( wrapper.html() ).toMatchSnapshot();
    });

    test('should show the hidden image and pokemon 100', () => {
        // Arrange
        const wrapper = shallowMount(PokemonImg, {
            props: {
                imgId: 100,
                showPokemon: false
            }
        });
        // Act
        const [ img1, img2 ] = wrapper.findAll('img');
        // Assert
        expect(img1.exists()).toBeTruthy();
        expect(img2).toBe(undefined);
        expect(img1.classes('hidden-pokemon')).toBeTruthy();
    });

    test('should show pokemon if showpicture is true', () => {
        
    });
});