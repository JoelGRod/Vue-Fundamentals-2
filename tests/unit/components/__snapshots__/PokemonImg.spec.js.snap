// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImg.vue should match with snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="fade-in" alt="Pokemon Image"></div>`;
